### PR TITLE
feat(bind-attr): add the ability to bind attributes

### PR DESCRIPTION
**Pull request by lgalfaso**
Add two new directives that bind expressions to attributes
once and on notify

### CODE REVIEW

File: src/directives/bind-attr-on-notify/bind-attr-on-notify.js [hunk 1]
```diff
@@ -0,0 +1,36 @@
...
+
+
+angular.module('fastBind.bindAttrOnNotify', []).
+  directive('fastBindAttrOnNotify', ['$parse', function($parse) {
+    var DEFAULT_EVENT_NAME = 'fast-bind-notify';
+
+    return {
+      compile: function compile(element, attributes) {
+        var expr = $parse(attributes.widgetsFastBindAttrsOnNotify),
+          name = attributes.fastBindOnNotifyName || DEFAULT_EVENT_NAME;
```
Comment (kseamon) [inline src/directives/bind-attr-on-notify/bind-attr-on-notify.js]: Trying to stick to the Google style guide, so please indent line continuations such as this by 4 spaces.

File: src/directives/bind-attr-on-notify/bind-attr-on-notify.js [hunk 2]
```diff
@@ -0,0 +1,36 @@
...
+        return function link(scope, element, attrs) {
+          var lastValues = {};
+          scope.$on(name, function() {
+            var values = expr(scope);
+
+            angular.forEach(values, function(value, key) {
+
+             if (value !== lastValues[key]) {
+              attrs.$set(key,value);
+            }
```
Comment (kseamon) [inline src/directives/bind-attr-on-notify/bind-attr-on-notify.js]: This line and the one above need +2 spaces indent

File: src/directives/bind-attr-on-notify/bind-attr-on-notify.js [hunk 3]
```diff
@@ -0,0 +1,36 @@
...
+
+        return function link(scope, element, attrs) {
+          var lastValues = {};
+          scope.$on(name, function() {
+            var values = expr(scope);
+
+            angular.forEach(values, function(value, key) {
+
+             if (value !== lastValues[key]) {
+              attrs.$set(key,value);
```
Comment (kseamon) [inline src/directives/bind-attr-on-notify/bind-attr-on-notify.js]: Add a space after the comma please.

File: src/directives/bind-attr-once/bind-attr-once.js [hunk 4]
```diff
@@ -0,0 +1,24 @@
...
+  directive('fastBindAttrOnce', ['$parse', function($parse) {
+    return {
+      compile: function compile(element, attributes) {
+        var expr = $parse(attributes.fastBindOnce);
+
+        return function link(scope, element) {
+          var values = expr(scope);
+
+          angular.forEach(values, function(value, key) {
+            attrs.$set(key,value);
```
Comment (kseamon) [inline src/directives/bind-attr-once/bind-attr-once.js]: Add a space after the comma please.

**Comment by kseamon** [comment]
Thanks for the PR!  Just a few small style comments and then I'll merge it.

**Comment by lgalfaso** [comment]
@kseamon Thanks, updated the patch and rebase

File: src/directives/bind-attr-on-notify/bind-attr-on-notify.js (hunk 1 above)
Comment (kseamon) [inline src/directives/bind-attr-on-notify/bind-attr-on-notify.js]: You got everything but this one. Please make it:

```
var expr = $parse(attributes.widgetsFastBindAttrsOnNotify),
    name = attributes.fastBindOnNotifyName || DEFAULT_EVENT_NAME;
```

**Comment by kseamon** [comment]
Added one more comment (expanding upon one of the original ones).

**Comment by lgalfaso** [comment]
Pushed a new patch with the style fix

**Comment by kseamon** [comment]
Merged. Thanks again!

**Pull request merged**